gear menu: Add organization info.

ganpa3 opened the pull request:
Fixes #22205

<details>
<summary><strong>Self-hosted users</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/195146764-5fe90b03-8b1d-4b1b-b879-ecc5c00312ea.png)

![image](https://user-images.githubusercontent.com/58626718/195146980-3c141711-5d2c-46d0-9313-ce07a737547e.png)

</details>


<details>
<summary><strong>Zulip Cloud Free users</strong></summary>

### Business orgs

![image](https://user-images.githubusercontent.com/58626718/195146017-b51b8759-0969-4b09-ab6f-99443527c864.png)

![image](https://user-images.githubusercontent.com/58626718/195145953-fef9c753-f8f3-4648-92a2-b262a9ba2c7d.png)

### Education orgs

![image](https://user-images.githubusercontent.com/58626718/195145455-31b26db7-33e6-4235-a5ed-6b55de149d7d.png)

![image](https://user-images.githubusercontent.com/58626718/195145333-2ba1409c-3119-4490-973a-e0631c5f7c65.png)

### All other orgs

![image](https://user-images.githubusercontent.com/58626718/195145722-3690677b-1fec-4a07-bff0-dd1b2842323f.png)

![image](https://user-images.githubusercontent.com/58626718/195145806-6f8fcbf4-e464-44ec-b98d-d9214e5c06b2.png)

</details>


<details>
<summary><strong>For Zulip Cloud Standard users</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/195147617-7b68403b-9776-40fd-9eca-8a4f2acbcece.png)

![image](https://user-images.githubusercontent.com/58626718/195147553-a4613b77-aca4-43db-b81a-87ec70b36d2d.png)

</details>


<details>
<summary><strong>For sponsored Zulip Cloud Standard users</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/195147835-055a87fd-1459-4ed4-9f93-6f5b004b1447.png)

![image](https://user-images.githubusercontent.com/58626718/195147889-6557865d-31e6-4dc0-9994-ea8606a2a391.png)

</details>

<!---
Fixes #22205

<details>
<summary><strong>Self-hosted users</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/193016341-d296fda1-ebb6-4acb-8837-24966217a7e2.png)

![image](https://user-images.githubusercontent.com/58626718/193016249-a583f57c-9849-44fb-8294-6ec6d552bee8.png)

### Long version name
![image](https://user-images.githubusercontent.com/58626718/193016643-6bb9c464-0c63-47eb-baf4-b0d060fc23ba.png)

![image](https://user-images.githubusercontent.com/58626718/193016720-7e1bea84-a247-4e62-b637-741e1f5c491f.png)

</details>


<details>
<summary><strong>Self-hosted users that need to upgrade (12px font-size)</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/193017863-0e6cc7d2-a5dc-437e-a14c-7b65a6a40c9d.png)

![image](https://user-images.githubusercontent.com/58626718/193017783-be42a01d-13d9-4c40-8c88-a7698809565d.png)

</details>


<details>
<summary><strong>Self-hosted users that need to upgrade (14px font-size)</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/193018139-4ddf0af7-0f89-48f4-b109-32634b608a07.png)

![image](https://user-images.githubusercontent.com/58626718/193018201-358f1a8a-cd58-4193-8118-da218a78eac1.png)

</details>


<details>
<summary><strong>Zulip Cloud Free users (12px font-size)</strong></summary>

### Business orgs

![image](https://user-images.githubusercontent.com/58626718/193019420-d76d7107-24f3-473f-bb4f-2ba4c92db518.png)

![image](https://user-images.githubusercontent.com/58626718/193019340-f5629823-05d6-43e6-8cf4-cc8813193656.png)

### Education orgs

![image](https://user-images.githubusercontent.com/58626718/193019599-050ed17c-2439-4cce-a4f4-943c7f4e751f.png)

![image](https://user-images.githubusercontent.com/58626718/193019633-d9207281-eeae-443d-bcfa-e06a21332e24.png)

### All other orgs

![image](https://user-images.githubusercontent.com/58626718/193020605-90b14f5f-5045-4f23-8b78-7b72c482b673.png)

![image](https://user-images.githubusercontent.com/58626718/193020535-5aba3703-f54c-4af1-a7f1-b5bec2bdaa67.png)

</details>


<details>
<summary><strong>Zulip Cloud Free users (14px font-size)</strong></summary>

### Business orgs

![image](https://user-images.githubusercontent.com/58626718/193031183-3244155a-1c1c-4aa0-a1dc-4e304ffa1d15.png)

![image](https://user-images.githubusercontent.com/58626718/193031096-919083d3-6f88-44be-a8db-3b5279adbd0a.png)

### Education orgs

![image](https://user-images.githubusercontent.com/58626718/193031430-82aabeb2-0e0e-4b4e-87dd-221c6596e970.png)

![image](https://user-images.githubusercontent.com/58626718/193031495-245948b4-0483-49fa-8046-c75b1f26fad4.png)

### All other orgs

![image](https://user-images.githubusercontent.com/58626718/193021117-e0d2b313-04c6-46a9-b85c-f51e4fab8f35.png)

![image](https://user-images.githubusercontent.com/58626718/193021177-b94b8a6f-df90-4fa4-80a2-33a20a117f5f.png)

</details>


<details>
<summary><strong>For Zulip Cloud Standard users</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/193031964-b5cc5e84-69b6-4f7c-864c-4f6ff087bcb8.png)

![image](https://user-images.githubusercontent.com/58626718/193031887-6833da7c-a2b2-4da3-95c0-4a7759382d76.png)

</details>


<details>
<summary><strong>For sponsored Zulip Cloud Standard users</strong></summary>

![image](https://user-images.githubusercontent.com/58626718/193032156-9735cb55-51ea-4bcc-be31-30ec7982625b.png)

![image](https://user-images.githubusercontent.com/58626718/193032231-61c14b94-2534-4f8f-abda-a5f0a2a98e9b.png)

</details>
-->

timabbott commented:
The "Upgrade to the latest release" text should probably have a link, right?

timabbott commented on static/templates/gear_menu.hbs:
```diff
@@ -8,6 +8,30 @@
+            <li class="org-info org-url">{{realm_uri}}</li>
+            <li class="org-info org-plan hidden-for-spectators">
+                {{#if is_plan_limited }}
+                    <a href="/plans" target="_blank" rel="noopener noreferrer" role="menuitem">Zulip Cloud Free</a>
```
@alya should these plan name strings be tagged for translation? Since the landing pages aren't translated, it's a bit unclear to me whether this bit should be.

alya replied:
Good question! As also noted below, maybe we should keep the plan names in English.

timabbott commented on static/templates/gear_menu.hbs:
```diff
@@ -8,6 +8,30 @@
+            {{/if}}
+            {{#if (and is_self_hosted server_needs_upgrade)}}
+            <li class="org-info">
+                <a href="https://zulip.readthedocs.io/en/stable/production/upgrade-or-modify.html" target="_blank" rel="noopener noreferrer" role="menuitem">Upgrade to the<br /> latest release</a>
```
Is the `br` here necessary, or could be instead set CSS for how it should line-wrap? 

The `br` hackery will not work if these need to be tagged for translation.

ganpa3 replied:
Removed `br` and used CSS `white-space`.

timabbott commented on static/templates/gear_menu.hbs:
```diff
@@ -8,6 +8,30 @@
+                {{else if is_plan_standard_sponsored_for_free}}
+                    <a href="/plans" target="_blank" rel="noopener noreferrer" role="menuitem">Zulip Cloud Standard,<br />sponsored for free</a>
+                {{else if is_self_hosted}}
+                    Version {{zulip_version}}
```
Certainly this one should be tagged for translation.

ganpa3 replied:
Done.

ganpa3 commented:
> The "Upgrade to the latest release" text should probably have a link, right?

It does have a link.
https://github.com/zulip/zulip/pull/22637/files#diff-0045645a6356fc9e36d9e580833ddbdeea7dbe8bfd995d71d4557fb53e82f85dR33

timabbott commented on static/templates/gear_menu.hbs:
```diff
@@ -8,6 +8,32 @@
+            </li>
+            {{#if (and is_plan_limited is_owner) }}
+            <li class="org-info">
+                <a href="/upgrade" target="_blank" rel="noopener noreferrer" role="menuitem">Upgrade to: Zulip Cloud Standard</a>
```
What if we make "Zulip Cloud Standard" be a `plan_name` variable that is passed into these strings for translation? @alya for thoughts on whether that's the right way to do i18n, given that we don't translate the plan names anywhere else.

alya replied:
Hm, we should perhaps mark "Upgrade to" for translation, but leave the plan names in English?

timabbott replied:
That's what I suggested -- with the `{plan_name}` part in a variable, so that folks can still do language-appropriate word order for the part that isn't the plan name.

timabbott commented:
A few notes:
* Clicking anywhere that isn't a link on the gear menu seems to close it; we probably want to remove that behavior, since it makes it hard to, say, copy the version number. 
* The hover behavior for the version number seems off; it turns to black text when you hover, which maybe makes it feels like there's some interaction there, but there doesn't seem to be one.
* I wonder if the "Version" line should be "Zulip server x.y"; I don't like "Version" being capitalized, and it's not clear what's version is being documented. Though more precise would be "Zulip server x.y", to distinguish from the desktop app's version, which we might want to display on an adjacent line if you're using the desktop app in a future revision (You can probably parse it out of user-agent, which is ugly).

As a sidenote, we should definitely remove the Bootstrap `nav` tab nonsense, and plan to migrate the gear menu to be a Tippy popover, using the framework defined in `static/js/popover_menus.js`, rather than using a Bootstrap popover; the gear menu has no Tippy tooltips, so isn't going to get blocked from that by the issues we had with tooltips on top of tooltips..

ganpa3 commented:
> Clicking anywhere that isn't a link on the gear menu seems to close it; we probably want to remove that behavior, since it makes it hard to, say, copy the version number.

The popover is removed by Bootstrap. How should we handle this?
https://github.com/zulip/zulip/blob/4391bc324b5f737659ee32790c64424cc3df80ed/static/third/bootstrap/js/bootstrap.js#L471-L473
> The hover behavior for the version number seems off; it turns to black text when you hover, which maybe makes it feels like there's some interaction there, but there doesn't seem to be one.

Fixed.
> I wonder if the "Version" line should be "Zulip server x.y"; I don't like "Version" being capitalized, and it's not clear what's version is being documented. Though more precise would be "Zulip server x.y", to distinguish from the desktop app's version, which we might want to display on an adjacent line if you're using the desktop app in a future revision (You can probably parse it out of user-agent, which is ugly).

Changed to `Zulip server`

alya commented:
Nice! A few questions/comments on the visuals/interactions:

1. I agree with Tim that we need to keep the menu open on non-link clicks.
2. Since the server version number is not a link, it should be in regular font color, not blue.
3. Let's remove the colon after "Upgrade to"
4. Could you please post a few more screenshots, since I'm not sure how to test this myself in dev? In particular, I'd like to see:
   a. A server than needs to be upgraded
   b. A server running off `main` and/or some custom branch, i.e. some situations where the version number is long and complicated

alya commented:
For Zulip Cloud Free orgs, we should probably mention sponsorship for non-business orgs, same as we do in the upgrade banners (cf. #22161 / #22172):
<img width="591" alt="Screen Shot 2022-08-22 at 9 21 54 AM" src="https://user-images.githubusercontent.com/2090066/185970441-d8d1efbc-b5a9-484b-840f-b7962777bb42.png">

In the context of a menu, maybe that just means putting another option just below "Upgrade to Zulip Cloud Standard". We could try:

- Business type orgs: Don't show anything
- For education orgs (profit or non-profit): "Request education pricing" (which matches the wording on https://zulip.com/for/education/)
- All other orgs, including unspecified: "Request sponsorship"

@sahil839 I'm now thinking we should add the "request education pricing" condition to the upgrade banner as well.

alya commented:
About the blue color for links, I'm currently no sure we should use it at all. We'll potentially have several links in the top section, and it attracts a lot of attention. Also, we currently don't use it for other options that are external links, like "Help center" and "API documentation". @terpimost what do you think?

alya commented:
I'm thinking about how to display the Zulip Cloud Standard sponsored org info -- the screenshots above don't look quite right to me.

First, a data question -- is it correct that the orgs we'd be showing to are "sponsored for free", or does the data we're using include subsidized orgs? Probably ideally we'd show something both to fully sponsored and partially sponsored orgs.

timabbott commented:
> First, a data question -- is it correct that the orgs we'd be showing to are "sponsored for free", or does the data we're using include subsidized orgs? Probably ideally we'd show something both to fully sponsored and partially sponsored orgs.

Correct, it's the `STANDARD_FREE` plan.

We don't have an easy way to check which organizations are subsidized; there is a `discount` percent field on their plan, but there are in theory multiple reasons a plan could be discounted.

I think it's reasonable to just do this design change with showing nothing special for subsidized organizations, and open a follow-up issue to address that once we're happy with the rest; the default behavior of showing "Zulip Cloud Standard" for discounted organizations is probably OK.

> About the blue color for links, I'm currently no sure we should use it at all. We'll potentially have several links in the top section, and it attracts a lot of attention. Also, we currently don't use it for other options that are external links, like "Help center" and "API documentation".

Yeah, I agree the blue links probably attract undue attention; the plan information is for most users just a nice informational detail.

@ganpa3 it'd be great if you can address the feedback you're able to; it feels like this is getting close to being possible to integrate.

timabbott commented:
> The popover is removed by Bootstrap. How should we handle this?

We can probably override this somehow. However, I think the best option might be to change this into component from a Bootstrap "dropdown" into a popover, since we're trying to get rid of Bootstrap anyway. The best option would be to migrate this menu to use `static/js/popover_menus.js`.

It's possible that will prove easiest to do as a quick prep PR, so we can test the no-functionality-changes migration to our modern popover library. But in any case, I think doing so should be a pretty simple PR, since the gear menu itself at present only has a few links. @amanagr is a good resource if you need any help.

I think it's worth prioritizing this over your other projects just to get it finished.

ganpa3 commented:
> Since the server version number is not a link, it should be in regular font color, not blue.

Done.
> Let's remove the colon after "Upgrade to"

Done.
> Could you please post a few more screenshots, since I'm not sure how to test this myself in dev?

Updated the PR description.
> For Zulip Cloud Free orgs, we should probably mention sponsorship for non-business orgs

Done.
> About the blue color for links, I'm currently no sure we should use it at all.

Removed.

ganpa3 commented:
I will create a separate PR for converting the gear menu to use `popover_menus.js`

timabbott commented:
For the "long version name" example, it'd be best to use the version number we have on chat.zulip.org -- that's the typical format for a self-hosted server. I'm not sure having the whole menu go really wider is better than having it be cut off with ellipses.

timabbott commented:
Other than that, I'm happy with everything about the styling except the "Upgrade to ..." text, which feels too long/big. Maybe the right idea is to go to 80% font size for it, not sure; I also think the line spacing feels off. we could also consider merging this without the "Upgrade to ..." lines and trying to fine tune that as a follow-up if it feels hard to get right.

ganpa3 commented:
> Other than that, I'm happy with everything about the styling except the "Upgrade to ..." text, which feels too long/big. Maybe the right idea is to go to 80% font size for it, not sure; I also think the line spacing feels off. we could also consider merging this without the "Upgrade to ..." lines and trying to fine tune that as a follow-up if it feels hard to get right.

@alya thoughts?

ganpa3 commented:
> For the "long version name" example, it'd be best to use the version number we have on chat.zulip.org

Can you please elaborate?
The version number here is the same as in the `About Zulip` modal.
https://github.com/zulip/zulip/blob/28173cafc86c531fe52b5403cba3e756181b4517/static/js/about_zulip.js#L24
If we hide the long version name with an ellipsis, one wouldn't be able to copy it.

alya commented:
> If we hide the long version name with an ellipsis, one wouldn't be able to copy it.

Mm, yeah... Could we line-wrap instead?

alya commented:
> > Other than that, I'm happy with everything about the styling except the "Upgrade to ..." text, which feels too long/big. Maybe the right idea is to go to 80% font size for it, not sure; I also think the line spacing feels off. we could also consider merging this without the "Upgrade to ..." lines and trying to fine tune that as a follow-up if it feels hard to get right.
> 
> @alya thoughts?

I agree with Tim's comment; let's try reducing font size on the "Upgrade to.." line. Can you post a couple of variants at different sizes?

For the ones that have the "Request sponsorship" / "Request education pricing" line, can we also try putting an "or" in there, maybe italicized on its own line in between? And let's make the spacing between those lines much tighter, to make it clear that they go together.

alya commented:
If it helps, we can focus on the self-hosted part of this PR, and split the Zulip Cloud changes into a follow-up. But I'd love to get the self-hosted changes merged for 6.0.

timabbott commented:
> The version number here is the same as in the About Zulip modal.

I think what I meant is that for screenshots, you should use the chat.zulip.org version string, rather than the development environment one, because development environments have shorter version number strings.

ganpa3 commented:
> > If we hide the long version name with an ellipsis, one wouldn't be able to copy it.
> 
> Mm, yeah... Could we line-wrap instead?

Done.

ganpa3 commented:
> For the ones that have the "Request sponsorship" / "Request education pricing" line, can we also try putting an "or" in there, maybe italicized on its own line in between? And let's make the spacing between those lines much tighter, to make it clear that they go together.

Done.

> I agree with Tim's comment; let's try reducing font size on the "Upgrade to.." line. Can you post a couple of variants at different sizes?

Posted screenshots and updated the PR description.

alya commented:
Maybe post in #design for faster iteration, since this is a release goal, so we'd like to merge it very soon? Thanks!

alya commented:
The latest screenshots [posted on CZO](https://chat.zulip.org/#narrow/stream/101-design/topic/UI.20redesign.3A.20gear.20menu/near/1446680) look good to me. Ready for your review, @timabbott !

timabbott commented:
Test-deploying this on chat.zulip.org.

timabbott commented:
I made a ton of changes to the logic for what strings to display when, both to handle additional cases better, and also to just make use of the visual space. The test suite you wrote was super helpful in being able to do this safely.

``` diff
diff --git a/frontend_tests/node_tests/gear_menu.js b/frontend_tests/node_tests/gear_menu.js
index ade181e9f2..d6c1f6ea5a 100644
--- a/frontend_tests/node_tests/gear_menu.js
+++ b/frontend_tests/node_tests/gear_menu.js
@@ -11,20 +11,57 @@ const gear_menu = zrequire("gear_menu");
 run_test("version_display_string", () => {
     let expected_version_display_string;
 
-    page_params.zulip_version = "Zulip server 6.0";
-    expected_version_display_string = "Zulip Server 6.0";
+    // An official release
+    page_params.zulip_version = "5.6";
+    page_params.zulip_merge_base = "5.6";
+    expected_version_display_string = "translated: Zulip Server 5.6";
     assert.equal(gear_menu.version_display_string(), expected_version_display_string);
 
-    page_params.zulip_version = "Zulip server 6.0-dev-1976-g4bb381fc80";
-    expected_version_display_string = "Zulip Server development branch";
+    // An official beta
+    page_params.zulip_version = "6.0-beta1";
+    page_params.zulip_merge_base = "6.0-beta1";
+    expected_version_display_string = "translated: Zulip Server 6.0-beta1";
     assert.equal(gear_menu.version_display_string(), expected_version_display_string);
 
-    page_params.zulip_version = "Zulip server 6.0-dev-1994-g93730766b0";
-    page_params.zulip_merge_base = "Zulip server 6.0-dev-1976-g4bb381fc80";
-    expected_version_display_string = "Fork of Zulip Server development branch";
+    // An official release candidate
+    page_params.zulip_version = "6.0-rc1";
+    page_params.zulip_merge_base = "6.0-rc1";
+    expected_version_display_string = "translated: Zulip Server 6.0-rc1";
     assert.equal(gear_menu.version_display_string(), expected_version_display_string);
 
-    page_params.zulip_version = "Zulip server 6.0+git-4-g385a408be5";
-    expected_version_display_string = "Fork of Zulip Server 6.0";
+    // The Zulip development environment
+    page_params.zulip_version = "6.0-dev+git";
+    page_params.zulip_merge_base = "6.0-dev+git";
+    expected_version_display_string = "translated: Zulip Server dev environment";
+    assert.equal(gear_menu.version_display_string(), expected_version_display_string);
+
+    // A commit on Zulip's main branch.
+    page_params.zulip_version = "6.0-dev-1976-g4bb381fc80";
+    page_params.zulip_merge_base = "6.0-dev-1976-g4bb381fc80";
+    expected_version_display_string = "translated: Zulip Server 6.0-dev";
+    assert.equal(gear_menu.version_display_string(), expected_version_display_string);
+
+    // A fork with 18 commits beyond Zulip's main branch.
+    page_params.zulip_version = "6.0-dev-1994-g93730766b0";
+    page_params.zulip_merge_base = "6.0-dev-1976-g4bb381fc80";
+    expected_version_display_string = "translated: Zulip Server 6.0-dev (modified)";
+    assert.equal(gear_menu.version_display_string(), expected_version_display_string);
+
+    // A commit from the Zulip 5.x branch
+    page_params.zulip_version = "5.6+git-4-g385a408be5";
+    page_params.zulip_merge_base = "5.6+git-4-g385a408be5";
+    expected_version_display_string = "translated: Zulip Server 5.6 (patched)";
+    assert.equal(gear_menu.version_display_string(), expected_version_display_string);
+
+    // A fork with 3 commits beyond the Zulip 5.x branch.
+    page_params.zulip_version = "5.6+git-4-g385a408be5";
+    page_params.zulip_merge_base = "5.6+git-7-abcda4235c2";
+    expected_version_display_string = "translated: Zulip Server 5.6 (modified)";
+    assert.equal(gear_menu.version_display_string(), expected_version_display_string);
+
+    // A fork of a Zulip release commit, not on 5.x branch.
+    page_params.zulip_version = "5.3-1-g7ed896c0db";
+    page_params.zulip_merge_base = "5.3";
+    expected_version_display_string = "translated: Zulip Server 5.3 (modified)";
     assert.equal(gear_menu.version_display_string(), expected_version_display_string);
 });
diff --git a/static/js/gear_menu.js b/static/js/gear_menu.js
index 7f2b046876..870f7f5691 100644
--- a/static/js/gear_menu.js
+++ b/static/js/gear_menu.js
@@ -101,18 +101,32 @@ export function update_org_settings_menu_item() {
 export function version_display_string() {
     const version = page_params.zulip_version;
     const is_fork = page_params.zulip_merge_base && page_params.zulip_merge_base !== version;
-    if (version.includes("dev")) {
-        if (is_fork) {
-            return "Fork of Zulip Server development branch";
-        }
-        return "Zulip Server development branch";
+
+    if (page_params.zulip_version.endsWith("-dev+git")) {
+        // The development environment uses this version string format.
+        return $t({defaultMessage: "Zulip Server dev environment"});
     }
 
-    const [major_version_number, minor_version_number] = version.match(/\d+/g).slice(0, 2);
     if (is_fork) {
-        return `Fork of Zulip Server ${major_version_number}.${minor_version_number}`;
+        // For forks, we want to describe the Zulip version this was
+        // forked from, and that it was modified.
+        const display_version = page_params.zulip_merge_base
+            .replace(/\+git.*/, "")
+            .replace(/-dev.*/, "-dev");
+        return $t({defaultMessage: "Zulip Server {display_version} (modified)"}, {display_version});
+    }
+
+    // The below cases are all for official versions; either a
+    // release, or Git commit from one of Zulip's official branches.
+
+    if (version.includes("+git")) {
+        // A version from a Zulip official maintenance branch such as 5.x.
+        const display_version = version.replace(/\+git.*/, "");
+        return $t({defaultMessage: "Zulip Server {display_version} (patched)"}, {display_version});
     }
-    return `Zulip Server ${major_version_number}.${minor_version_number}`;
+
+    const display_version = version.replace(/\+git.*/, "").replace(/-dev.*/, "-dev");
+    return $t({defaultMessage: "Zulip Server {display_version}"}, {display_version});
 }
 
 export function initialize() {
diff --git a/static/templates/gear_menu.hbs b/static/templates/gear_menu.hbs
index 21c434c3a2..b603fa045b 100644
--- a/static/templates/gear_menu.hbs
+++ b/static/templates/gear_menu.hbs
@@ -12,7 +12,7 @@
             <li class="org-info org-url">{{realm_uri}}</li>
             {{#if is_self_hosted }}
                 <li class="org-info org-version">
-                    <a href="#about-zulip" role="menuitem">{{#tr}}{version_display_string}{{/tr}}</a>
+                    <a href="#about-zulip" role="menuitem">{{version_display_string}}</a>
                 </li>
                 {{#if server_needs_upgrade }}
                 <li class="org-info org-upgrade small-font-size">
```

timabbott commented:
With those changes, which also fix the i18n issues I'd noticed with this pull request, I think this is ready to merge.
